sql-plugin/src/main/scala/com/nvidia/spark/rapids/RmmRapidsRetryIterator.scala

```diff
@@ -580,9 +581,14 @@ object RmmRapidsRetryIterator extends Logging {
             lastException = ex
 
             if (!topLevelIsRetry && !causedByRetry) {
-              // we want to throw early here, since we got an exception
-              // we were not prepared to handle
-              throw lastException
+              // If the exception is the result of a CUDF column size overflow, attempt split-retry.
+              ex match {
```

Comment:
The complexity of `isRetryOrSplitAndRetry` and `causedByRetryOrSplit` is somewhat to deal with cases where someone ends up wrapping the exception. Do we want to do the same thing here? or combine it with the existing checks.

Reply:
Now that you mention it, it appears that `causedByRetryOrSplit` checks if `RetryOOM` or `SplitAndRetryOOM` appear anywhere in the exception stack.

Combining the overflow check into `causedByRetryOrSplit` might obfuscate the function, both in its name and in the return values. I've used a separate function instead.